### PR TITLE
Add test framework

### LINKED ISSUE #11: Add Test-Environment Builder
Add a test-environment builder that launches LightWalletd and ZCashD/ZebraD automatically for integration tests.

This is required for CI testing to be implemented.

Prerequisite of #10, #12.

### LINKED ISSUE #14: Implement wallet-side Nym service RPCs on CompactTxStreamerClient and expose Nym Address
Implement wallet-side Nym service RPCs on CompactTxStreamerClient and expose Nym Address. Also allow Nym/Standard implementation to be chosen in code.

- Provides a better interface for wallets to use this functionality and for testing.

### PR DESCRIPTION
**Pull request by idky137**
Adds test code to allow integration tests to be run without a manual launch of zcashd and lightwalletd (required for adding integration tests to CI). Restructures Nym Code and removes feature gate.

- uses Zingolib's RegtestManager to launch Lightwalletd and Zcashd in regtest mode for integration tests.
- uses Zingolib's ClientBuilder to create a test wallet.
- adds proxy launcher for use in testing. Upgrades zingoproxyd using this.
- adds test manager to launch RegtestManager and Zingo-Proxy test server.
- simplifies feature use.
- (moves Nym enhanced CompactTxStreamerServer service RPC code to zingo-rpc::rpc::nymservice)

Closes #11 

Note: CI will be added/fixed in a separate PR.

### CODE REVIEW

File: zingo-proxyd/src/server.rs [hunk 1]
```diff
@@ -2,37 +2,47 @@
...
         tokio::task::spawn(async move {
             let svc = CompactTxStreamerServer::new(self.0);
             let sockaddr = SocketAddr::new(std::net::IpAddr::V4(Ipv4Addr::LOCALHOST), port.into());
-            println!("Proxy listening on {sockaddr}");
-            tonic::transport::Server::builder()
-                .add_service(svc)
-                .serve(sockaddr)
-                .await
+            println!("GRPC server listening on: {sockaddr}");
+            while online.load(Ordering::SeqCst) {
```
Comment (AloeareV) [inline zingo-proxyd/src/server.rs]: This wants to be a select!, not a while.

**Review by AloeareV — APPROVED** [review]
Other than the while vs select! issue we've discussed, this all looks awesome!

File: zingo-proxyd/src/server.rs (hunk 1 above)
Comment (idky137) [reply zingo-proxyd/src/server.rs]: updated to use tokio::select! in [5ee32c5].

File: zingo-proxyd/src/bin/zingoproxyd.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@ use std::{
 
 use zingoproxylib::proxy::spawn_proxy;
 
-extern crate ctrlc;
+use ctrlc;
```
Comment (AloeareV) [inline zingo-proxyd/src/bin/zingoproxyd.rs]: Still unneeded, ctrlc is the crate name so you can path directly from it anyway.

**Review by AloeareV — APPROVED** [review]

File: zingo-proxyd/src/bin/zingoproxyd.rs (hunk 2 above)
Comment (idky137) [reply zingo-proxyd/src/bin/zingoproxyd.rs]: removed in e824800

**Pull request merged**